lib/connectors/base/connector.rb

```diff
@@ -13,25 +13,25 @@
 module Connectors
   module Base
     class Connector
-      def yield_documents(connector_settings); end
-
-      def source_status(params = {})
-        health_check(params)
-        { :status => 'OK', :statusCode => 200, :message => "Connected to #{display_name}" }
-      rescue StandardError => e
-        { :status => 'FAILURE', :statusCode => e.is_a?(custom_client_error) ? e.status_code : 500, :message => e.message }
+      def self.display_name
+        raise 'Not implemented for this connector'
       end
 
-      def display_name
-        raise 'Not implemented for this connector'
+      def self.configurable_fields
+        {}
       end
 
       def service_type
         self.class::SERVICE_TYPE
```

Comment:
This is confusing. Why not make it a static method `service_type` in each connector, and remove the Constant `SERVICE_TYPE`?

Reply:
Makes sense, will do that!

Reply:
Fixed in https://github.com/elastic/connectors/pull/170/commits/2902f4fde1d3e5006887567b8f0b0c10800793a0

Reply:
I'm ambivalent, but just wanted to share...

> Why not make it a static method service_type in each connector, and remove the Constant SERVICE_TYPE?

Adam Harwood was always a fan of using functions over constants. I asked him about this once, and he explained that it's a lot more clear to override a function in a child class than it is to override a constant. I thought this was an interesting pattern, as I hadn't seen it before. 

:homerdisappear: